app/src/main/java/com/example/music_app/MainActivityViewModel.kt

```diff
@@ -27,13 +31,21 @@ class MainActivityViewModel(
 
     private fun checkStatus() {
         viewModelScope.launch {
-            isAuthorizedCheckUseCase()
-                .collect {isAuth ->
-                    _route.value = if (isAuth) Screens.PlaylistsScreen else Screens.LoginScreen
-                    delay(WAIT_TIME)
-                    _isSplashScreenVisible.value = false
+            val isAuth: Boolean = isAuthorizedCheckUseCase().component1()?: false
+            when {
+                !isAuth -> _route.value = Screens.LoginScreen
+                isOutdatedCheckUseCase().component1()?: false -> {
+                    requestRefreshTokenUseCase().collect { token ->
+                        if (token.component1() != null) {
+                            _route.value =
+                                Screens.PlaylistsScreen
+                        }
+                    }
                 }
+                else -> _route.value = Screens.PlaylistsScreen
```

Comment:
I think the usage of component1 invocation could be replaced with the usage of `onSuccess` function